src/nnue/features/full_threats.cpp

```diff
@@ -243,8 +254,11 @@ void FullThreats::append_changed_indices(Square           ksq,
                                          IndexList&       added,
                                          FusedUpdateData* fusedData,
                                          bool             first) {
-    for (const auto dirty : diff.list)
+    const int orientation = OrientTBL[Perspective][ksq];
+
+    for (std::size_t idx = 0; idx < diff.list.size(); ++idx)
     {
+        const DirtyThreat& dirty = diff.list[idx];
```

Comment:
We can still use a range-based for loop here.
```suggestion
    for (const auto& dirty : diff.list)
    {
```